src/movies.js

```diff
@@ -1,28 +1,59 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(mArray) {
+  const allDirectors = mArray.map(function (direct) {
+    return direct.director;
+  });
+  return allDirectors;
+}
+getAllDirectors(movies);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arrMov) {
+  let numbOfMov = arrMov.filter(function (movie) {
+    if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama'))
+      return movie;
+
+  });
+  return (numbOfMov.length);
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage() {
+
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore() { }
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(movieYear) {
+  const result = movieYear.sort((firstScore, secondScore) => {
```

Comment:
This iteration was tricky, because the task asks to return a new array.
Sort method changes the existing array, so it means firstly you have to make a copy of your original array, and only then sort() that new array:

function orderByYear(movieYear) {
    // clone your array:
       const result = 'you make copy of your movieYear array here'
   
    // sort your new array
       result.sort((firstScore, secondScore) => {})


couple of options to clone an array:
-> spread operator:
     const newArray = [...originalArray]
-> slice():
     const newArray = originalArray.slice()